src/opening-period/time-span/TimeSpan.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect } from 'react';
+import { Button as HDSButton, IconTrash, Select, TextInput } from 'hds-react';
+import Weekdays from './Weekdays';
+import './TimeSpan.scss';
+import TimeInput from './TimeInput';
+import { ResourceStateOption } from '../../common/lib/types';
+
+export default function TimeSpan({
+  index,
+  remove,
+  register,
+  setValue,
+  resourceStateOptions,
+}: {
+  index: number;
+  remove: Function;
+  register: Function;
+  setValue: Function;
+  resourceStateOptions: ResourceStateOption[] | undefined;
+}): JSX.Element {
+  useEffect(() => {
+    register({ name: `timeSpans[${index}].resourceState` });
+  });
+
+  if (!resourceStateOptions) {
+    throw new Error('This should never happen but typescript');
+  }
+
+  return (
+    <div data-test={`time-span-${index}`} className="time-span-container">
+      <div className="time-span-first-header-row">
+        <Weekdays index={index} register={register} />
+        <HDSButton
```

Comment:
Can this be replaced with our button implementation?

Reply:
At the moment, no. Because our button only supports primary and secondary types. We could make a separate supplementary button, but because this was the first time when it was needed, I did not yet refactor it to a separate component, to avoid abstraction. But once we have the need for it, I think it should be extracted